redfish-core/lib/oem/ibm/usb_code_update.hpp

```diff
@@ -0,0 +1,152 @@
+        std::vector<std::pair<std::string, MapperServiceMap>>;
+
+    crow::connections::systemBus->async_method_call(
+        [aResp, path,
+         handler{std::move(handler)}](const boost::system::error_code ec,
+                                      const MapperGetSubTreeResponse& subtree) {
+        std::string service{};
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR << "DBUS response error, ec: " << ec.value();
+            messages::internalError(aResp->res);
+            return;
+        }
+
+        for (const auto& [objectPath, serviceMap] : subtree)
+        {
+            if (objectPath != path)
+            {
+                continue;
+            }
+
+            if (serviceMap[0].first.empty() || serviceMap.size() != 1)
+            {
+                BMCWEB_LOG_ERROR << "usb code update mapper error!";
+                messages::internalError(aResp->res);
+                return;
+            }
+
+            handler(serviceMap[0].first);
+            return;
+        }
+
+        BMCWEB_LOG_DEBUG << "Can't find usb code update service!";
+        return;
+        },
+        "xyz.openbmc_project.ObjectMapper",
+        "/xyz/openbmc_project/object_mapper",
+        "xyz.openbmc_project.ObjectMapper", "GetSubTree",
+        "/xyz/openbmc_project", 0,
+        std::array<const char*, 1>{
+            "xyz.openbmc_project.Control.Service.Attributes"});
+}
+
+/**
+ * @brief Retrieves BMC USB code update state.
+ *
+ * @param[in] aResp     Shared pointer for generating response message.
+ *
+ * @return None.
+ */
+inline void
+    getUSBCodeUpdateState(const std::shared_ptr<bmcweb::AsyncResp>& aResp)
+{
+    BMCWEB_LOG_DEBUG << "Get USB code update state";
+    auto callback = [aResp](const std::string& service) {
+        crow::connections::systemBus->async_method_call(
+            [aResp](const boost::system::error_code ec,
+                    std::variant<bool>& usbCodeUpdateState) {
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR << "DBUS response error " << ec;
+                messages::internalError(aResp->res);
+                return;
+            }
+
+            const bool* usbCodeUpdateStatePtr =
+                std::get_if<bool>(&usbCodeUpdateState);
+
+            if (!usbCodeUpdateStatePtr)
+            {
+                BMCWEB_LOG_ERROR << "Can't get USB code update status!";
+                messages::internalError(aResp->res);
+                return;
+            }
+            aResp->res.jsonValue["Oem"]["IBM"]["@odata.type"] =
+                "#OemManager.IBM";
+            aResp->res.jsonValue["Oem"]["IBM"]["@odata.id"] =
+                "/redfish/v1/Managers/bmc#/Oem/IBM";
+            aResp->res.jsonValue["Oem"]["IBM"]["USBCodeUpdateEnabled"] =
+                *usbCodeUpdateStatePtr;
+            },
+            service, usbCodeUpdateObjectPath, "org.freedesktop.DBus.Properties",
+            "Get", "xyz.openbmc_project.Control.Service.Attributes", "Enabled");

```

Comment:
All upstream bmcweb code is now using     `sdbusplus::asio::getProperty<bool>(`

https://github.com/openbmc/bmcweb/blob/3ccb3adb9a14783f6bef601506de9f8bcae22d51/redfish-core/lib/led.hpp#L39

Reply:
done.